### PR TITLE
Update LSF module to keep consistent with DPDispatcher

### PR DESCRIPTION
**Pull request by Cloudac7**
In this PR, the code structure of `LSF` class has been rewritten to be same as `PBS` and `Slurm`, from which only headers of scripts are defined. The change has been tested and the test passed.

### CODE REVIEW

File: dpdispatcher/lsf.py [hunk 1]
```diff
@@ -2,141 +2,44 @@
...
-                queue_name=resources.queue_name),
-            'lsf_walltime_line': "#BSUB -W {walltime}".format(
-                walltime=resources.kwargs.get('walltime', '12:00'))
+                queue_name=resources.queue_name)
         }
-        gpu_usage_flag = resources.kwargs.get('gpu_usage', False)
-        gpu_new_syntax_flag = resources.kwargs.get('gpu_new_syntax', False)
+        gpu_usage_flag = resources.kwargs['kwargs'].get('gpu_usage', False)
+        gpu_new_syntax_flag = resources.kwargs['kwargs'].get('gpu_new_syntax', False)
+        print(gpu_usage_flag)
```
Comment (njzjz) [inline dpdispatcher/lsf.py:40]: Is it for debug?

Comment (Cloudac7) [reply dpdispatcher/lsf.py]: Yes and the print message has been removed after the latest push. Sorry for not noticing. 
However, the both flags are parameters to control how to get GPU resources for LSF, which is not for debug.

**Review by njzjz — APPROVED** [review]

File: dpdispatcher/lsf.py [hunk 2]
```diff
@@ -2,141 +2,41 @@
...
             'lsf_partition_line': "#BSUB -q {queue_name}".format(
-                queue_name=resources.queue_name),
-            'lsf_walltime_line': "#BSUB -W {walltime}".format(
-                walltime=resources.kwargs.get('walltime', '12:00'))
+                queue_name=resources.queue_name)
         }
-        gpu_usage_flag = resources.kwargs.get('gpu_usage', False)
-        gpu_new_syntax_flag = resources.kwargs.get('gpu_new_syntax', False)
+        gpu_usage_flag = resources.kwargs['kwargs'].get('gpu_usage', False)
+        gpu_new_syntax_flag = resources.kwargs['kwargs'].get('gpu_new_syntax', False)
```
Comment (njzjz) [inline dpdispatcher/lsf.py:39]: We don't have documents for these two parameters. Maybe we need to discuss where to add documents first.

File: dpdispatcher/lsf.py [hunk 3]
```diff
@@ -2,141 +2,41 @@
...
-#         self.walltime = walltime
-#         self.gpu_new_syntax = gpu_new_syntax
-#         self.gpu_usage = gpu_usage
-#         self.prepend_text = prepend_text
-#         self.append_text = append_text
-#         self.lsf_bsub_dict = lsf_bsub_dict
 
-
-class LSF(Machine):
+class LSF(Machine, ABC):
```
Comment (njzjz) [inline dpdispatcher/lsf.py:20]: why is it abstract base class?

**Review by felix5572 — APPROVED** [review]

File: dpdispatcher/lsf.py (hunk 3 above)
Comment (Cloudac7) [reply dpdispatcher/lsf.py]: It is a PyCharm issue and I just pressed the button, let PyCharm solve it automatically. `ABC` has been removed here for it is not an abstract base class. And from `Machine`, I found two abstract methods not implemented by the derived class and added.

**Pull request merged**